opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
All this is making my brain hurt. Once upon a time, there were these violent email discussions about how links could never have a name. (Personally, I thought links should have a name, but was kicked and poked and stabbed until it hurt) So that is what got implemented, and that is a deep core assumption that is very deeply embedded in the design of absolutely everything.  If we suddenly allowed links to have names, we'd have to rethink the design of EvaluationLinks, which are central and used everywhere. Go back and redesign the forward/backward chainers, the pattern matcher.  Redisgn the way we do the linguistics processing. Redesign the SQL database internals.  Vast tracts of code would have to change.  So that change is imponderable at this time.

The disasterous disassembly for NodeCast suggests a major design flaw, and that, indeed, we should keep the virtual methods on Atom and try to avoid casts everywhere. Its shocking and I don't understand it just right now.